### PR TITLE
add ai_thinker_esp_12k_nodemcu

### PR DESCRIPTION
**Pull request by codetyphon**
add AI Thinker ESP-12K (ESP32-S2) NodeMCU Development Board

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -528,6 +528,7 @@ jobs:
         - "unexpectedmaker_feathers2"
         - "unexpectedmaker_feathers2_prerelease"
         - "unexpectedmaker_tinys2"
+        - "ai_thinker_esp_12k"
```
Comment (dhalbert) [inline .github/workflows/build.yml:531]: Please put this in alphabetical order.

File: ports/esp32s2/ai_thinker_esp_12k/mpconfigboard.h [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+
+// Micropython setup
+
+// Same setup as the Saola board but with no Neopixel on board
+
+#define MICROPY_HW_BOARD_NAME       "NodeMCU-32-S2 w/ESP-12K 08"
+#define MICROPY_HW_MCU_NAME         "ESP-12K 08"
```
Comment (dhalbert) [inline ports/esp32s2/ai_thinker_esp_12k/mpconfigboard.h:32]: Is this just the module or is the dev board?

File: ports/esp32s2/ai_thinker_esp_12k/mpconfigboard.mk [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+
+# The default queue depth of 16 overflows on release builds,
+# so increase it to 32.
+CFLAGS += -DCFG_TUD_TASK_QUEUE_SZ=32
+
+CIRCUITPY_ESP_FLASH_MODE=dio
+CIRCUITPY_ESP_FLASH_FREQ=40m
+CIRCUITPY_ESP_FLASH_SIZE=4MB
+
+CIRCUITPY_MODULE=wrover
```
Comment (dhalbert) [inline ports/esp32s2/ai_thinker_esp_12k/mpconfigboard.mk:17]: This file does not end in a newline.

**Review by dhalbert — CHANGES_REQUESTED** [review]
Let us know more about this. Are you using this board, or is it just the module, or are you representing AI-Thinker? Thanks.

File: ports/esp32s2/ai_thinker_esp_12k/mpconfigboard.h (hunk 2 above)
Comment (codetyphon) [reply ports/esp32s2/ai_thinker_esp_12k/mpconfigboard.h:32]: A Development Board base on esp32-s2

Comment (codetyphon) [reply ports/esp32s2/ai_thinker_esp_12k/mpconfigboard.h:32]: A Development Board base on esp32-s2

File: .github/workflows/build.yml (hunk 1 above)
Comment (codetyphon) [reply .github/workflows/build.yml]: i am a user use this dev board .    not AI-Thinker  。   iam Third party 。

File: .github/workflows/build.yml [hunk 4]
```diff
@@ -506,6 +506,7 @@ jobs:
         - "adafruit_metro_esp32s2"
         - "artisense_rd00"
         - "atmegazero_esp32s2"
+        - "ai_thinker_esp_12k"
```
Comment (dhalbert) [inline .github/workflows/build.yml:509]: Sorry, this is still not quite in alphabetical order. Put it before the artisense

File: ports/esp32s2/ai_thinker_esp_12k/mpconfigboard.h (hunk 2 above)
Comment (dhalbert) [reply ports/esp32s2/ai_thinker_esp_12k/mpconfigboard.h:32]: It's this, https://docs.ai-thinker.com/en/12k_development_board_esp32-s2, is that right? Sometimes it seems to be called "ESP-12K-KIT" or "ESP-12K NodeMCU". Could you pick what seems to be the best name and change the directory name to that (e.g. `ai_thinker_esp_12k_nodemcu`), and also change the strings here and in `mpconfigboard.mk`?

Another link: https://www.tinytronics.nl/shop/en/development-boards/microcontroller-boards/with-wi-fi/ai-thinker-nodemcu-32-s2-esp-12k

I just don't want the board to be confused with the bare module.

Also, I don't see the "08" designation anywhere. I don't see where that came from. If you don't need it, let's remove it.

File: ports/esp32s2/ai_thinker_esp_12k/mpconfigboard.mk [hunk 5]
```diff
@@ -0,0 +1,17 @@
+USB_VID = 0x612B
+USB_PID = 0x80A7
+USB_PRODUCT = "NodeMCU-32-S2 w/ESP-12K 08"
```
Comment (dhalbert) [inline ports/esp32s2/ai_thinker_esp_12k/mpconfigboard.mk:3]: Change name as necessary.

**Review by dhalbert — CHANGES_REQUESTED** [review]

File: ports/esp32s2/ai_thinker_esp_12k/mpconfigboard.h (hunk 2 above)
Comment (codetyphon) [reply ports/esp32s2/ai_thinker_esp_12k/mpconfigboard.h]: ok

File: ports/esp32s2/boards/ai_thinker_esp_12k_nodemcu/mpconfigboard.h [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+
+// Micropython setup
+
+// Same setup as the Saola board but with no Neopixel on board
+
+#define MICROPY_HW_BOARD_NAME       "ESP 12k NodeMCU"
+#define MICROPY_HW_MCU_NAME         "ESP-12K"
```
Comment (dhalbert) [inline ports/esp32s2/boards/ai_thinker_esp_12k_nodemcu/mpconfigboard.h:32]: Only one more thing. I missed this before, sorry. `MICROPY_HW_MCU_NAME` should be the name of the chip, not the module.
```suggestion
#define MICROPY_HW_MCU_NAME         "ESP32S2"
```

**Review by dhalbert — CHANGES_REQUESTED** [review]
Just one more change I missed before,sorry.

**Review by dhalbert — APPROVED** [review]
Thanks for your patience!

Could you now submit a PR to https://github.com/adafruit/circuitpython-org, to add this board to the downloads section in circuitpython.org? Details of what to submit are here:
https://github.com/adafruit/circuitpython-org#readme. Thank you.

**Comment by jepler** [comment]
@codetyphon Hello.

Can you please let me know how you chose the USB VID:PID when creating this PR?

**Pull request merged**